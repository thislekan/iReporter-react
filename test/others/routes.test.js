import '@babel/polyfill';
import React from 'react';
import setup from '../mock/setup';
import Routes from '../../src/routes/index.jsx';

describe('<Header />', () => {
  it('should render the route component', () => {
    const wrapper = setup(<Routes />);
    expect(wrapper).toMatchSnapshot();
  });
});
