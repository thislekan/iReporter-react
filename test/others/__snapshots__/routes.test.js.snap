// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render the route component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <AppRouter>
        <BrowserRouter>
          <Router
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
          >
            <div>
              <withRouter(Header)>
                <Route>
                  <Header
                    history={
                      Object {
                        "action": "POP",
                        "block": [Function],
                        "createHref": [Function],
                        "go": [Function],
                        "goBack": [Function],
                        "goForward": [Function],
                        "length": 1,
                        "listen": [Function],
                        "location": Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        },
                        "push": [Function],
                        "replace": [Function],
                      }
                    }
                    location={
                      Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      }
                    }
                    match={
                      Object {
                        "isExact": true,
                        "params": Object {},
                        "path": "/",
                        "url": "/",
                      }
                    }
                  >
                    <div>
                      <header>
                        <nav>
                          <div>
                            <div
                              style={
                                Object {
                                  "alignItems": "center",
                                  "display": "flex",
                                  "justifyContent": "center",
                                  "marginLeft": "10px",
                                }
                              }
                            >
                              <NavLink
                                className=""
                                exact={true}
                                to="/"
                              >
                                <Route
                                  exact={true}
                                  path="\\\\/"
                                >
                                  <Link
                                    aria-current="page"
                                    className="active"
                                    style={Object {}}
                                    to="/"
                                  >
                                    <a
                                      aria-current="page"
                                      className="active"
                                      href="/"
                                      onClick={[Function]}
                                      style={Object {}}
                                    >
                                      iReporter
                                    </a>
                                  </Link>
                                </Route>
                              </NavLink>
                            </div>
                            <div>
                              <NavLink
                                to="/login"
                              >
                                <Route
                                  path="\\\\/login"
                                >
                                  <Link
                                    aria-current={null}
                                    to="/login"
                                  >
                                    <a
                                      aria-current={null}
                                      href="/login"
                                      onClick={[Function]}
                                    >
                                      Login
                                    </a>
                                  </Link>
                                </Route>
                              </NavLink>
                            </div>
                          </div>
                        </nav>
                      </header>
                    </div>
                  </Header>
                </Route>
              </withRouter(Header)>
              <Switch>
                <Route
                  component={[Function]}
                  computedMatch={
                    Object {
                      "isExact": true,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                  exact={true}
                  location={
                    Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    }
                  }
                  path="/"
                >
                  <HomePage
                    history={
                      Object {
                        "action": "POP",
                        "block": [Function],
                        "createHref": [Function],
                        "go": [Function],
                        "goBack": [Function],
                        "goForward": [Function],
                        "length": 1,
                        "listen": [Function],
                        "location": Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        },
                        "push": [Function],
                        "replace": [Function],
                      }
                    }
                    location={
                      Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      }
                    }
                    match={
                      Object {
                        "isExact": true,
                        "params": Object {},
                        "path": "/",
                        "url": "/",
                      }
                    }
                  >
                    <div>
                      <section>
                        <div>
                          <div>
                            <h1>
                              You too can help
                              <span>
                                  
                              </span>
                              <span>
                                end
                              </span>
                               corruption.
                            </h1>
                            <button
                              id="go-to-signup"
                              onClick={[Function]}
                            >
                              Sign Up
                            </button>
                          </div>
                          <div>
                            <div>
                              <img
                                alt="transparent phone"
                                src={Object {}}
                              />
                              <img
                                alt="report incident screenshot"
                                src={Object {}}
                              />
                            </div>
                          </div>
                        </div>
                      </section>
                      <section>
                        <div>
                          <div>
                            <h3>
                              Lend the government your voice
                            </h3>
                          </div>
                          <div>
                            <div>
                              <img
                                alt="protester with megaphone"
                                src={Object {}}
                              />
                            </div>
                            <div
                              className=""
                            >
                              <p>
                                With an internet connection, you too can help fight corruption. All you need to do is register with us and report incidents with their location, leaving the rest to us
                              </p>
                              <p>
                                We make sure you remain anonymous while we investigate and you get real time notifications on your reorted incidents.
                              </p>
                            </div>
                          </div>
                        </div>
                      </section>
                      <section
                        className={[Function]}
                      >
                        <div>
                          <div
                            className=""
                          >
                            <h3>
                              How it works.
                            </h3>
                          </div>
                          <div>
                            <div>
                              <img
                                alt=""
                                src={Object {}}
                              />
                            </div>
                            <div
                              className=""
                            >
                              <p>
                                You too can fight corruption in 3 easy steps
                              </p>
                              <div>
                                <p>
                                  <span>
                                    1.
                                  </span>
                                   Sign up and verify your email.
                                </p>
                                <p>
                                  <span>
                                    2.
                                  </span>
                                   Login after verification.
                                </p>
                                <p>
                                  <span>
                                    3.
                                  </span>
                                   Report an incident.
                                </p>
                              </div>
                            </div>
                          </div>
                        </div>
                      </section>
                      <section>
                        <div>
                          <div>
                            <h3>
                              Our numbers have been good
                            </h3>
                          </div>
                          <div>
                            <div>
                              <div>
                                <span>
                                  All inicdents
                                </span>
                                <span
                                  className=""
                                >
                                  1000
                                </span>
                              </div>
                              <div>
                                <span>
                                  Resolved incidents
                                </span>
                                <span
                                  className=""
                                >
                                  550
                                </span>
                              </div>
                              <div>
                                <span>
                                  Rejected incidents
                                </span>
                                <span
                                  className=""
                                >
                                  200
                                </span>
                              </div>
                              <div>
                                <span>
                                  Under Investigation
                                </span>
                                <span
                                  className=""
                                >
                                  250
                                </span>
                              </div>
                            </div>
                            <div>
                              <p>
                                We pride ourselves in the numbers of resolved incidents, but we aim for even better results. Our team of investigators are thorough and discreet in their approach.
                              </p>
                              <p>
                                Every incident is treated as a priority. Our system and approach are void of bias, meaning we hold ourselves to a great standard. The numbers tell our stories for us.
                              </p>
                            </div>
                          </div>
                        </div>
                      </section>
                      <footer>
                        <div>
                          <p>
                            disclaimer
                          </p>
                          <p>
                            iReporter is a non governmental organization committed to ending corruption in nigeria
                          </p>
                        </div>
                        <div>
                          <p>
                            contact us 
                            <a
                              href="mailto:email@ireporter.net"
                            >
                              email@ireporter.net
                            </a>
                          </p>
                        </div>
                      </footer>
                    </div>
                  </HomePage>
                </Route>
              </Switch>
            </div>
          </Router>
        </BrowserRouter>
      </AppRouter>
    </Router>
  </BrowserRouter>
</Provider>
`;
